Add lsp-ocaml recipe

anmonteiro opened the pull request:
### Brief summary of what the package does

OCaml/Reason support for lsp-mode using [ocaml-language-server](https://github.com/freebroccolo/ocaml-language-server).

### Direct link to the package repository

https://github.com/emacs-lsp/lsp-ocaml

### Your association with the package

I'm the maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Thanks.
- Don't check `(featurep 'projectile)`. Instead check `projectile-mode` with `bound-and-true-p`: you shouldn't use projectile unless the user has enabled it.
- You're using `seq-some`, but that's not available in every Emacs > 24. Either specify a higher Emacs version, or declare a package dependency on the version of `seq` that is available in GNU ELPA.

anmonteiro commented:
@purcell thanks for the review. Just addressed your feedback in https://github.com/emacs-lsp/lsp-ocaml/commit/aa4240ef24b9eea578361c9d88b52b0138a925bf

purcell commented:
Awesome, thanks!
